[ET-VK][Ops] quantize ops skeleton test framework

ahmtox opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11553
* #11557
* #11483
* #11369
* #11556
* #11555
* #11554
* #11482
* #11481
* #11480
* #11368
* #11367
* __->__ #11366
* #11575
* #11365
* #11552
* #11479

Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

Commit:
[ET-VK][Ops] quantize ops skeleton test framework

Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

manuelcandales commented on backends/vulkan/test/op_tests/quantize_test.cpp:
```diff
@@ -0,0 +1,276 @@
+    int64_t quant_max,
+    at::ScalarType dtype) {
+  auto out = at::empty_like(input, dtype);
+  // Convert at::ScalarType to executorch::ScalarType
```
see comments on #11480

manuelcandales commented on backends/vulkan/test/op_tests/quantize_test.cpp:
```diff
@@ -0,0 +1,276 @@
+//
+
+// Helper function to get the name of a ScalarType for better error messages
+std::string scalar_type_name(c10::ScalarType dtype) {
```
see comments on #11480

manuelcandales commented on backends/vulkan/test/op_tests/quantize_test.cpp:
```diff
@@ -0,0 +1,276 @@
+  }
+}
+
+vkcompute::vkapi::ScalarType from_at_scalartype(c10::ScalarType at_scalartype) {
```
see comments on #11480

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] quantize ops skeleton test framework"


Skeleton framework that is needed to build out the quantize_per_tensor and quantize_per_token operators based on cpu implementation

Differential Revision: [D75959066](https://our.internmc.facebook.com/intern/diff/D75959066/)

[ghstack-poisoned]
